fibonacci/aws-fibonacci-java/src/main/java/com/serverless/Fibonacci.java

```diff
@@ -0,0 +1,88 @@
+package com.serverless;
+
+import java.io.IOException;
+import java.util.Map;
+
+import com.amazonaws.services.lambda.runtime.Context;
+import com.amazonaws.services.lambda.runtime.RequestHandler;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+public class Fibonacci implements RequestHandler<Map<String, Object>, ApiGatewayResponse> {
+	
+	@Override
+	public ApiGatewayResponse handleRequest(Map<String, Object> input, Context context) {
+		@SuppressWarnings("unchecked")
+		Map<String, String> queryParameters = (Map<String, String>) input.get("queryStringParameters");
+		String body = (String) input.get("body");
+		
+		String n_string = "";
+		if (queryParameters != null && queryParameters.containsKey("n")) {
+			n_string = queryParameters.get("n");
+		} else if (body != null) {
+			ObjectMapper mapper = new ObjectMapper();
+
+			try {
+				JsonNode bodyJSON = mapper.readTree(body);
+				
+				if (bodyJSON.has("n")) {
+					n_string = bodyJSON.get("n").asText();
+				}
+			} catch (IOException e) {
+				return createErrorResponse("Please pass valid JSON as body.");
+			}
+		}
+		
+		if (!isNumeric(n_string)) {
+			return createErrorResponse("Please pass a valid number 'n'.");
+	    }
+
+	    long n = Integer.parseInt(n_string);
+	    long result = fibonacci(n);
+	    
+	    return createSuccessResponse(String.valueOf(result));
+	}
+
+	/*@Override
+	public ApiGatewayResponse handleRequest(Map<String, String> parameters, Context context) {
+		String n_query = parameters.get("n_query");
```

Comment:
I found a simpler way to retrieve parameters in AWS (commented out) but it requires to define the interface of the function in a separate configuration. We can discuss which of them is better.